src/httpserver.cpp

```diff
@@ -480,6 +475,14 @@ void StopHTTPServer()
         workQueue->WaitExit();
         delete workQueue;
     }
+    if (eventBase) {
+        LogPrint("http", "Waiting for HTTP event thread to exit\n");
+        // Give event loop a few seconds to exit (to send back last RPC responses), then break it
+        if (!threadHTTP.try_join_for(boost::chrono::milliseconds(2000))) {
```

Comment:
@MarcoFalke but this doesn't stop new connections (or is that done elsewhere?) in that waiting time.  So conceptually, the status quo could be the same?